### ISSUE TITLE
[distributed] Higher allocations found in master branch v/s release branch for the same codepath

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio#3080]
## Expected Behavior

Comparable allocations should be present v/s release branch.

## Current Behavior

Significantly higher allocations for larger objects.

```
benchmark                                    old allocs     new allocs     delta
BenchmarkGetObjectVerySmallFS-12             27             644            +2285.19%
BenchmarkGetObject10KbFS-12                  28             648            +2214.29%
BenchmarkGetObject100KbFS-12                 28             651            +2225.00%
BenchmarkGetObject1MbFS-12                   28             656            +2242.86%
BenchmarkGetObject5MbFS-12                   56             658            +1075.00%
BenchmarkGetObject10MbFS-12                  84             659            +684.52%
BenchmarkGetObject25MbFS-12                  189            660            +249.21%
BenchmarkGetObject50MbFS-12                  346            661            +91.04%
BenchmarkGetObjectParallelVerySmallFS-12     26             636            +2346.15%
BenchmarkGetObjectParallel10KbFS-12          26             636            +2346.15%
BenchmarkGetObjectParallel100KbFS-12         26             636            +2346.15%
BenchmarkGetObjectParallel1MbFS-12           26             636            +2346.15%
BenchmarkGetObjectParallel5MbFS-12           52             636            +1123.08%
BenchmarkGetObjectParallel10MbFS-12          78             636            +715.38%
BenchmarkGetObjectParallel25MbFS-12          182            635            +248.90%
BenchmarkGetObjectParallel50MbFS-12          338            634            +87.57%
BenchmarkPutObjectPart5MbFS-12               16375          112007         +584.01%
BenchmarkPutObjectPart10MbFS-12              7211           54935          +661.82%
BenchmarkPutObjectPart25MbFS-12              3862           28414          +635.73%
BenchmarkPutObjectPart50MbFS-12              2478           15496          +525.34%
BenchmarkPutObjectVerySmallFS-12             56             2018           +3503.57%
BenchmarkPutObject10KbFS-12                  56             2019           +3505.36%
BenchmarkPutObject100KbFS-12                 56             2035           +3533.93%
BenchmarkPutObject1MbFS-12                   56             2035           +3533.93%
BenchmarkPutObject5MbFS-12                   90             2036           +2162.22%
BenchmarkPutObject10MbFS-12                  124            2040           +1545.16%
BenchmarkPutObject25MbFS-12                  261            2764           +959.00%
BenchmarkPutObject50MbFS-12                  466            3501           +651.29%
BenchmarkParallelPutObjectVerySmallFS-12     56             3263           +5726.79%
BenchmarkParallelPutObject10KbFS-12          56             3266           +5732.14%
BenchmarkParallelPutObject100KbFS-12         56             3288           +5771.43%
BenchmarkParallelPutObject1MbFS-12           56             3311           +5812.50%
BenchmarkParallelPutObject5MbFS-12           94             3314           +3425.53%
BenchmarkParallelPutObject10MbFS-12          130            3310           +2446.15%
BenchmarkParallelPutObject25MbFS-12          273            4047           +1382.42%
```
## Possible Solution

There is no possible right now 

## Steps to Reproduce (for bugs)

`go test -run=NONE -bench Bench.*FS`
## Context

Higher allocations can lead to slugging code path, hot spots and overall memory usage.

**Comment by vadmeste** [comment]
@harshavardhana, there was a regression which executed bench in XL instead of FS. @hackintoshrao just submitted a new PR (thanks!)

This is the new result:

```
benchmark                                   old allocs     new allocs     delta
BenchmarkGetObjectVerySmallFS-4             27             39             +44.44%
BenchmarkGetObject10KbFS-4                  28             40             +42.86%
BenchmarkGetObject100KbFS-4                 28             40             +42.86%
BenchmarkGetObject1MbFS-4                   28             44             +57.14%
BenchmarkGetObject5MbFS-4                   57             96             +68.42%
BenchmarkGetObject10MbFS-4                  84             157            +86.90%
BenchmarkGetObject25MbFS-4                  189            338            +78.84%
BenchmarkGetObject50MbFS-4                  346            639            +84.68%
BenchmarkGetObjectParallelVerySmallFS-4     26             36             +38.46%
BenchmarkGetObjectParallel10KbFS-4          26             37             +42.31%
BenchmarkGetObjectParallel100KbFS-4         26             37             +42.31%
BenchmarkGetObjectParallel1MbFS-4           26             37             +42.31%
BenchmarkGetObjectParallel5MbFS-4           52             86             +65.38%
BenchmarkGetObjectParallel10MbFS-4          78             147            +88.46%
BenchmarkGetObjectParallel25MbFS-4          182            326            +79.12%
BenchmarkGetObjectParallel50MbFS-4          338            624            +84.62%
BenchmarkPutObjectPart5MbFS-4               16387          18423          +12.42%
BenchmarkPutObjectPart10MbFS-4              7218           9126           +26.43%
BenchmarkPutObjectPart25MbFS-4              3866           5322           +37.66%
BenchmarkPutObjectPart50MbFS-4              2488           3776           +51.77%
BenchmarkPutObjectVerySmallFS-4             56             127            +126.79%
BenchmarkPutObject10KbFS-4                  56             127            +126.79%
BenchmarkPutObject100KbFS-4                 56             127            +126.79%
BenchmarkPutObject1MbFS-4                   56             132            +135.71%
BenchmarkPutObject5MbFS-4                   90             184            +104.44%
BenchmarkPutObject10MbFS-4                  124            254            +104.84%
BenchmarkPutObject25MbFS-4                  261            464            +77.78%
BenchmarkPutObject50MbFS-4                  466            815            +74.89%
BenchmarkParallelPutObjectVerySmallFS-4     56             126            +125.00%
BenchmarkParallelPutObject10KbFS-4          56             126            +125.00%
BenchmarkParallelPutObject100KbFS-4         56             128            +128.57%
BenchmarkParallelPutObject1MbFS-4           56             130            +132.14%
BenchmarkParallelPutObject5MbFS-4           90             185            +105.56%
BenchmarkParallelPutObject10MbFS-4          125            254            +103.20%
BenchmarkParallelPutObject25MbFS-4          264            464            +75.76%
```

**Comment by harshavardhana** [comment]
Fixed by #3146
